Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/StartSceneSystem.cs

```diff
@@ -0,0 +1,71 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using Cysharp.Threading.Tasks;
+using ECS.Abstract;
+using ECS.SceneLifeCycle.Components;
+using Ipfs;
+using Newtonsoft.Json;
+using SceneRunner;
+using System.Linq;
+using System.Threading;
+using UnityEngine;
+
+namespace ECS.SceneLifeCycle.Systems
+{
+    [UpdateInGroup(typeof(SimulationSystemGroup))]
+    [UpdateAfter(typeof(LoadSceneSystem))]
+    public partial class StartSceneSystem : BaseUnityLoopSystem
+    {
+        private readonly ISceneFactory sceneFactory;
+
+        private readonly IIpfsRealm ipfsRealm;
+
+        public StartSceneSystem(World world, IIpfsRealm ipfsRealm, ISceneFactory sceneFactory) : base(world)
+        {
+            this.sceneFactory = sceneFactory;
+            this.ipfsRealm = ipfsRealm;
+        }
+
+        protected override void Update(float t)
+        {
+            ProcessSceneToLoadQuery(World);
+        }
+
+        private async UniTask InitializeSceneAndStart(IpfsTypes.SceneEntityDefinition sceneDefinition, CancellationToken ct)
+        {
+            // main thread
+            var sceneFacade = await sceneFactory.CreateSceneFromSceneDefinition(ipfsRealm, sceneDefinition, ct);
+
+            ct.RegisterWithoutCaptureExecutionContext(() =>
+            {
+                sceneFacade?.DisposeAsync();
+            });
+
+            // thread pool
+            await sceneFacade.StartUpdateLoop(30, ct);
+        }
+
+        [Query]
+        [All(typeof(SceneLoadingComponent))]
+        private void ProcessSceneToLoad(in Entity entity, ref SceneLoadingComponent sceneLoadingComponent)
+        {
+            // If the scene just spawned, we start the request
+            sceneLoadingComponent.CancellationTokenSource = new CancellationTokenSource();
+            var cts = sceneLoadingComponent.CancellationTokenSource;
+
+            var liveSceneComponent = new LiveSceneComponent()
+            {
+                CancellationTokenSource = cts,
+            };
+
+            World.Add(entity, liveSceneComponent);
+
+            sceneLoadingComponent.Request = InitializeSceneAndStart(sceneLoadingComponent.Definition, sceneLoadingComponent.CancellationTokenSource.Token);
+
+            World.Remove<SceneLoadingComponent>(entity);
```

Comment:
Please review this function: you assign some data to `SceneLoadingComponent` and then immediately remove it, something is definitely wrong